core/src/test/java/kafka/test/ClusterTestExtensionsTest.java

```diff
@@ -48,14 +51,18 @@ public class ClusterTestExtensionsTest {
 
     // Static methods can generate cluster configurations
     static void generate1(ClusterGenerator clusterGenerator) {
-        clusterGenerator.accept(ClusterConfig.defaultClusterBuilder().name("Generated Test").build());
+        Map<String, String> serverProperties = new HashMap<>();
+        serverProperties.put("foo", "bar");
+        clusterGenerator.accept(ClusterConfig.defaultBuilder()
+                .setName("Generated Test")
+                .setServerProperties(serverProperties)
+                .build());
     }
 
     // BeforeEach run after class construction, but before cluster initialization and test invocation
     @BeforeEach
     public void beforeEach(ClusterConfig config) {
```

Comment:
As `ClusterConfig` becomes immutable object, developers can modify the configs by injection of `ClusterConfig`. Maybe we should remove this usage from code base.  WDYT?

Reply:
Make sense to me. Thanks

Reply:
Could you please file a ticket for it?

Reply:
filed https://issues.apache.org/jira/browse/KAFKA-16627